### PR TITLE
Added optional snippets and badges #31 #30 #42

### LINKED ISSUE #30: Add the ability to include badges within generated documentation
Documentation relating to DevOps processes commonly includes links for badges.

See [quickstart templates](https://github.com/Azure/azure-quickstart-templates/tree/master/modules/Microsoft.KeyVault/vaults/1.0)

### LINKED ISSUE #31: Allow parameter file snippet to be disabled
Allow generation of snippet to be disabled for the template.

### LINKED ISSUE #42: Bump PSDocs dependency to v0.8.0
Update PSDocs dependency to latest version.

### PR DESCRIPTION
**Pull request by BernieWhite**
## PR Summary

- New features:
  - Added the ability to disable parameter file snippet. #31
    - To disable parameter file snippet set configuration `AZURE_USE_PARAMETER_FILE_SNIPPET`.
    - See about_PSDocs_Azure_Configuration for details.
  - Added the ability to include badges in template document. #30
    - Set the `.ps-docs/azure-template-badges.md` file to include badge content.
    - See about_PSDocs_Azure_Badges for details.
- Engineering:
  - Bump PSDocs dependency to v0.8.0. #42

Fixes #31 
Fixes #30 
Fixes #42

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**

### CODE REVIEW

File: docs/concepts/en-US/about_PSDocs_Azure_Badges.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+For example:
+
+```markdown
+![label](https://image_uri)
+```
+
+To include badges images with a clickable link use standard markdown syntax:
+
+```markdown
+[![label](https://image_uri)](https://link_uri)
```
Comment (vicperdana) [inline docs/concepts/en-US/about_PSDocs_Azure_Badges.md:37]: Can we use the standard status Az DevOps badge such as [Build Status?

File: docs/concepts/en-US/about_PSDocs_Azure_Badges.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+The contents of this file is automatically inserted in generated output after the title but before description.
+
+### Include badges
+
+To include badge images use standard markdown syntax within the `azure-template-badges.md` file.
+For example:
+
+```markdown
+![label](https://image_uri)
```
Comment (vicperdana) [inline docs/concepts/en-US/about_PSDocs_Azure_Badges.md:31]: Maybe applicable to add a pre-reqs that the badge image must be publicly accessible?

**Review by vicperdana — APPROVED** [review]
Added some feedback for considerations.

File: docs/concepts/en-US/about_PSDocs_Azure_Badges.md (hunk 1 above)
Comment (BernieWhite) [reply docs/concepts/en-US/about_PSDocs_Azure_Badges.md:37]: Nice suggestion.

File: docs/concepts/en-US/about_PSDocs_Azure_Badges.md (hunk 2 above)
Comment (BernieWhite) [reply docs/concepts/en-US/about_PSDocs_Azure_Badges.md:31]: Doesn't need to be accessible publicly as long as the user has permissions. Good call out, I'll add something in.

**Pull request merged**